packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -89,49 +127,68 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
         }
         if (!regex.test(userInput)) {
           errorMessageToDisplay = strings.invalidInputErrorMessage;
+        } else if (timeRange) {
+          const optionDate: Date = getDateFromTimeSelection(useHour12, dateStartAnchor, userInput);
+          if (optionDate < dateStartAnchor || optionDate > dateEndAnchor) {
+            errorMessageToDisplay = format(
+              strings.timeOutOfBoundsErrorMessage,
+              dateStartAnchor.toString(),
+              dateEndAnchor.toString(),
+            );
+          }
         }
         return errorMessageToDisplay;
       };
 
-      const key = option?.key;
-      let updatedUserText = '';
       let errorMessageToDisplay = '';
-      if (value) {
+      if (input) {
         if (allowFreeform && !option) {
           if (!onFormatDate) {
             // Validate only if user did not add onFormatDate
-            errorMessageToDisplay = validateUserInput(value);
+            errorMessageToDisplay = validateUserInput(input);
           } else {
             // Use user provided validation if onFormatDate is provided
             if (onValidateUserInput) {
-              errorMessageToDisplay = onValidateUserInput(value);
+              errorMessageToDisplay = onValidateUserInput(input);
             }
           }
         }
-        updatedUserText = value;
-      } else if (option) {
-        updatedUserText = option.text;
       }
 
-      if (onChange && !errorMessageToDisplay) {
-        const selectedTime = value || option?.text || '';
-        const date = getDateFromTimeSelection(useHour12, baseDate, selectedTime);
-        onChange(event, date);
+      if (onGetErrorMessage) {
+        onGetErrorMessage(errorMessageToDisplay);
+      }
```

Comment:
What is this callback achieving? It seems to ignore anything returned from the callback?

Reply:
Ah yes, so I thought I would supply a callback prop `onGetErrorMessage` to TimePicker just in case users want to validate the component and then, for example, disable a button due to TimePicker validation failure. Please let me know if this sounds okay with you.

Reply:
Hmm in that case, it sounds like it might better be called `onError`. The current name implies that it's getting the message from the callback.

`onError` should pass in the native event as its first parameter, and a `data` object with the error message as its second parameter. The data object allows for passing additional parameters in the future.

```ts
type OnErrorData = {
  errorMessage?: string;
};
```
```ts
  onError?: (event: React.FormEvent<IComboBox>, data: OnErrorData) => void;
```

And then call it similar to `onChange`:
```ts
onError?.(ev, { errorMessage: errorMessageToDisplay });
```

Reply:
Resolved by creating `onValidationError` prop based on suggestion.

Reply:
I modified the suggestion a bit. Instead of the suggested
```javascript
if (errorMessageToDisplay && onError) {
  onError(ev, { errorMessage: errorMessageToDisplay });
}}
```
I wrote
```javascript
if (onValidationError) {
  onValidationError(ev, { errorMessage: errorMessageToDisplay });
}}
```.
I think the onValidationError callback prop must always be run, even if there is no error message, so that users that are listening to the `errorMessage` field are able to reset their validation result if there was a validation error that gets subsequently resolved.